lib/cocoapods/installer/xcode/pods_project_generator.rb

```diff
@@ -170,109 +177,121 @@ def install_libraries
+              [pod_target.name, target_installer.install!]
+            end]
 
-            aggregate_targets.sort_by(&:name).each do |target|
-              target_installer = AggregateTargetInstaller.new(sandbox, project, target)
-              target_installer.install!
+            # Hook up system framework dependencies for the pod targets that were just installed.
+            pod_target_installation_result_values = pod_target_installation_results.values.compact
+            unless pod_target_installation_result_values.empty?
+              add_system_framework_dependencies(pod_target_installation_result_values)
             end
 
-            add_system_framework_dependencies
+            aggregate_target_installation_results = Hash[aggregate_targets.sort_by(&:name).map do |target|
+              target_installer = AggregateTargetInstaller.new(sandbox, @project, target)
+              [target.name, target_installer.install!]
+            end]
+
+            return [pod_target_installation_results, aggregate_target_installation_results]
           end
         end
 
-        def integrate_targets
-          pod_targets_to_integrate = pod_targets.select { |pt| !pt.test_native_targets.empty? || pt.contains_script_phases? }
-          unless pod_targets_to_integrate.empty?
+        def integrate_targets(pod_target_installation_results)
+          pod_installations_to_integrate = pod_target_installation_results.values.select do |pod_target_installation_result|
+            pod_target = pod_target_installation_result.target
+            !pod_target_installation_result.test_native_targets.empty? || pod_target.contains_script_phases?
+          end
+          unless pod_installations_to_integrate.empty?
             UI.message '- Integrating targets' do
-              pod_targets_to_integrate.each do |pod_target|
-                PodTargetIntegrator.new(pod_target).integrate!
+              pod_installations_to_integrate.each do |pod_target_installation_result|
+                PodTargetIntegrator.new(pod_target_installation_result).integrate!
               end
             end
           end
         end
 
-        def add_system_framework_dependencies
-          # @TODO: Add Specs
-          pod_targets.select(&:should_build?).sort_by(&:name).each do |pod_target|
-            test_file_accessors, file_accessors = pod_target.file_accessors.partition { |fa| fa.spec.test_specification? }
-            file_accessors.each do |file_accessor|
-              add_system_frameworks_to_native_target(file_accessor, pod_target.native_target)
-            end
-            test_file_accessors.each do |test_file_accessor|
-              native_target = pod_target.native_target_for_spec(test_file_accessor.spec)
-              add_system_frameworks_to_native_target(test_file_accessor, native_target)
+        def add_system_framework_dependencies(pod_target_installation_results)
+          pod_target_installation_results.each do |target_installation_result|
```

Comment:
This was previously sorted by name, changing that will cause project diffs through having a different sort order for system frameworks. (Not necessarily a big deal, but it's a behaviour change)

Reply:
Also looks like this fn is missing test coverage

Reply:
Will fix the sorting and yes test coverage was *always* missing from this method, I had removed the "TODO add specs" above. Will add tests.

Reply:
test added. This was easier as the `OrangeFramework` pod I use depends on `s.frameworks 'UIKIt'` so I wrote one that ensures the native target includes this framework.

Reply:
Fixed sorting